_posts/2021/2021-02-22-url-helpers-outside-views-controllers.md

```diff
@@ -0,0 +1,89 @@
+---
+title: "Use Rails URL helpers outside view and controllers"
+description: "Sometimes useful to have in a job or model"
+layout: article
+category: ruby
+image:
+  base: "2021/url-helpers-outside-views-controllers"
+  alt: "Abstract greyscale roads"
+  source: "https://unsplash.com/photos/36b7JBzhfF4"
+  credit: "Bogdan Karlenko"
+---
+
+Occasionally you'll need to generate a URL for your own application outside of the normal places; views or controllers.
```

Comment:
Do you mean outside of "the normal places, views, or controllers"...or "outside of the normal places, which are views or controllers"?

Either way, I'd update the way the list is written, and drop the semicolon.

Reply:
```suggestion
Occasionally you'll need to generate a URL for your own application outside views or controllers where your URL helpers are available. These helper methods are [generated form your routes](https://guides.rubyonrails.org/routing.html#path-and-url-helpers), e.g. `user_books_path(user)`.
```